pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -878,27 +856,82 @@ def list(self, request, *args, **kwargs):
         """
         Returns paginated CollectionVersions list.
         """
+
+        # there are too many potential permutations of query params to be able
+        # to cache all of them, so this will only cache queries that stick to
+        # limit/offset params, which the ansible-galaxy client uses
+        cache_request = True
+        for k in request.query_params.keys():
+            if k not in ("offset", "limit"):
+                cache_request = False
+                break
+
+        cache_key = "".join(
+            [
+                str(self._repository_version.pk),
+                kwargs["namespace"],
+                kwargs["name"],
+                str(request.query_params.get("offset", "0")),
+                str(request.query_params.get("limit", "0")),
+            ]
+        )
+
+        if cache_key in cache and cache_request:
+            return Response(cache.get(cache_key))
+
         queryset = self.get_queryset()
 
         # prevent OOMKILL ...
-        queryset = queryset.only("pk", "content_ptr_id")
+        queryset = queryset.only(
+            "pk",
+            "content_ptr_id",
+            "marks",
+            "namespace",
+            "name",
+            "version",
+            "pulp_created",
+            "pulp_last_updated",
+            "requires_ansible",
+            "collection",
+        )
 
         queryset = self.filter_queryset(queryset)
 
-        # This is -very- slow when the collection has many versions.
-        # queryset = sorted(
-        #    queryset, key=lambda obj: semantic_version.Version(obj.version), reverse=True
-        # )
-
```

Comment:
Do we not want to bring back sorting by version now that version has been broken up? Right now I think the returned results are default sorted by `pulp_created` which will typically be the same as sorted versions.

Reply:
sorting by versions will still likely be slower, so I'm not sure it's worth bringing this back since it seems to be working fine.